models/repo.go

```diff
@@ -1583,14 +1584,22 @@ func SearchRepositoryByName(opts *SearchRepoOptions) (repos []*Repository, _ int
 	}
 
 	repos = make([]*Repository, 0, opts.PageSize)
-
-	// Append conditions
-	sess := x.Where("LOWER(lower_name) LIKE ?", "%"+opts.Keyword+"%")
-	if opts.OwnerID > 0 {
-		sess.And("owner_id = ?", opts.OwnerID)
+	sess := x.Alias("repo")
+	// Attempt to find repositories that opts.UserId has access to
+	// This does not include other people's private repositories even if opts.UserId is an admin
+	if !opts.Private && opts.UserID > 0 {
+		sess.Join("LEFT", []string{"access", "acc"}, "acc.repo_id = repo.id")
```

Comment:
1. Is there a particular reason you use `LEFT` not `INNER` for join?
2. Rename `access` to `acc` is over simplifying, just use `access` is fine and obvious.